### PR TITLE
Add boxcar_radius for manual extraction

### PR DESCRIPTION
**Pull request by profxj**
As titled.

Fixes what I perceived to be a bug in
that no manual extraction would be performed
unless *at least* one "real" source was detected.

Also fixes a link issue for the new `bspline` code.

Testing is underway.

### CODE REVIEW

**Comment by tepickering** [comment]
the test failures are due to the release of `tox` 4.0.0. this has stuffed a lot of people's CI. i'm looking into it to see if there's an easy fix we can do...

**Comment by tepickering** [comment]
well, i ran tox tests locally and they worked fine, but pulled in `tox=-4.0.1` now. so i re-ran the workflow to see if they do the same.

**Comment by tepickering** [comment]
nope, same problem.

**Comment by tepickering** [comment]
this looks like a problem:
`test: install_package> python -I -m pip install --force-reinstall --no-deps /home/runner/work/PypeIt/PypeIt/.tox/.tmp/package/1/pypeit-0.1.dev1+g15c1d0a.tar.gz`
i get the right version in the filename when i run `tox` locally. 

update: this is a red herring and an artifact of the runner doing a depth=1 fetch so it doesn't have the full tag history.

**Comment by tepickering** [comment]
ok, whatever was plaguing github actions yesterday has been fixed. the tests now run and the current failures are real.

File: pypeit/core/findobj_skymask.py [hunk 1]
```diff
@@ -1244,146 +1244,153 @@ def objs_in_slit(image, ivar, thismask, slit_left, slit_righ,
...
+            # Instantiate a null specobj and return
+            msgs.info('No objects found automatically.  Consider manual extraction.')
+            return sobjs
         else:
-            nperslit_bpm = np.zeros(npeaks_all, dtype=bool)
+            msgs.info('No objects found automatically.')
+            
     else:
-        nperslit_bpm = np.zeros(npeaks_all, dtype=bool)
+        # Compute the formal corresponding variance over the set of pixels that are not masked by gpm_sigclip
```
Comment (kbwestfall) [inline pypeit/core/findobj_skymask.py:1258]: We don't necessarily have to do so for this PR, but this function really needs to be broken up.  There are now more than 100 lines of code in this `else` block.

File: pypeit/core/findobj_skymask.py [hunk 2]
```diff
@@ -1244,146 +1244,153 @@ def objs_in_slit(image, ivar, thismask, slit_left, slit_righ,
...
+            # assign FWHM to all these objects
+            if use_user_fwhm:
+                sobjs[iobj].FWHM = fwhm
+            else:
+                sobjs[iobj].FWHM = get_fwhm(fwhm, nsamp, sobjs[iobj].smash_peakflux, sobjs[iobj].SPAT_FRACPOS, flux_smash_smth)
+
+            # assign BOX_RADIUS
+            sobjs[iobj].BOX_RADIUS = boxcar_rad
+
+        if len(sobjs) == 0 and hand_extract_dict is None:
```
Comment (kbwestfall) [inline pypeit/core/findobj_skymask.py:1371]: Nice change here!

File: pypeit/manual_extract.py [hunk 3]
```diff
@@ -96,15 +103,22 @@ def by_fitstbl_input(cls, frame: str, inp: str, spectrograph):
             idict['spat'] += [float(parse[1])]
             idict['spec'] += [float(parse[2])]
             idict['fwhm'] += [float(parse[3])]
+            # Boxcar?
+            if len(parse) >= 5:
+                idict['boxcar_rad'] += [float(parse[4])]
+            else:
+                idict['boxcar_rad'] += -1.
```
Comment (kbwestfall) [inline pypeit/manual_extract.py]: Should this be `idict['boxcar_rad'] += [-1.]`; i.e., you're appending to the list not adding a value, right?

**Review by kbwestfall — APPROVED** [review]
Nice!  Just a few comments from me, including one possible bug.  Approving now, though.

**Comment by tepickering** [comment]
test failures:
`=========================== short test summary info ============================`
`FAILED ../../.tox/test-gingadev/lib/python3.8/site-packages/pypeit/tests/test_manual.py::test_instantiate`
`FAILED ../../.tox/test-gingadev/lib/python3.8/site-packages/pypeit/tests/test_manual.py::test_dict_for_obj`

File: pypeit/core/findobj_skymask.py (hunk 1 above)
Comment (profxj) [reply pypeit/core/findobj_skymask.py:1258]: Agreed, and I considered it a bit.

But the entire extraction codes need refactoring.  
And that's a bigger job than this PR..

File: pypeit/manual_extract.py (hunk 3 above)
Comment (profxj) [reply pypeit/manual_extract.py]: yeeps, good catch!

Comment (profxj) [reply pypeit/manual_extract.py]: Added a test to catch this from here on.

**Review by badpandabear — APPROVED** [review]

**Review by tepickering — APPROVED** [review]
with tests passing, looks good to me.

**Comment by profxj** [comment]
DevSuite passed, but am going to re-run with all the merges into `develop`

Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  205 passed, 708 warnings in 230.16s (0:03:50) ---
--- PYTEST UNIT TESTS PASSED  118 passed, 1238 warnings in 975.58s (0:16:15) ---
--- PYTEST VET TESTS PASSED  31 passed, 979 warnings in 811.54s (0:13:31) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 155/155 TESTS  ---
Testing Started at 2022-12-19T17:19:34.883847
Testing Completed at 2022-12-20T04:23:37.803797
Total Time: 11:04:02.919950

**Comment by profxj** [comment]
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  205 passed, 62 warnings in 385.05s (0:06:25) ---
--- PYTEST UNIT TESTS PASSED  118 passed, 210 warnings in 999.63s (0:16:39) ---
--- PYTEST VET TESTS PASSED  29 passed, 10 warnings in 863.68s (0:14:23) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 155/155 TESTS  ---
Testing Started at 2022-12-27T23:08:20.926935
Testing Completed at 2022-12-28T16:53:30.402790
Total Time: 17:45:09.475855

**Pull request merged**